compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -305,21 +305,28 @@ pub(crate) fn create_query_frame<
     name: &'static str,
 ) -> QueryStackFrame {
     // Avoid calling queries while formatting the description
-    let description = ty::print::with_no_queries!(do_describe(tcx, key));
+    let description = ty::print::with_no_queries!(
+        // Disable visible paths printing for performance reasons.
+        // Showing visible path instead of any path is not that important in production.
+        ty::print::with_no_visible_paths!(
+            // Force filename-line mode to avoid invoking `type_of` query.
+            ty::print::with_forced_impl_filename_line!(do_describe(tcx, key))
+        )
+    );
     let description = if tcx.sess.verbose_internals() {
         format!("{description} [{name:?}]")
     } else {
         description
     };
-    let span = if kind == dep_graph::dep_kinds::def_span {
+    let span = if kind == dep_graph::dep_kinds::def_span || with_no_queries() {
```

Comment:
Would bringing only these last 2 changes back as `with_reduced_queries()` fix the issue?

Reply:
I opened https://github.com/rust-lang/rust/pull/121993 for that. It fixes OOM for the test case we have.

Reply:
nice, thanks!